modules/libcom/src/error/errlog.c

```diff
@@ -476,8 +477,22 @@ int errlogSetConsole(FILE *stream)
 {
     errlogInit(0);
     epicsMutexMustLock(pvt.msgQueueLock);
+    if (pvt.closeConsole) {
+        fclose(pvt.console);
+    }
     pvt.console = stream ? stream : stderr;
     pvt.ttyConsole = isATTY(pvt.console);
+#ifdef vxWorks
+    /* The stderr/stdout we get may become invalid as they are thread specific.
+     * Also, the FDs are actually maps to some "real" FDs.
+     * Thus, we clone the pvt.console and link it with a dup of the "real" FD.
+     * The dup'ed FD will be closed when the FILE handle is fclosed. */
+    if (fileno(pvt.console) <= 2) {
```

Comment:
I think this condition should be checked earlier, before overwriting `pvt.console`.  Doing so gives a simpler path on error.  I expect that `fdopen()`, `dup()`, and perhaps `ioTaskStdGet()` can fail.

Also, would a more direct test for the thread specific case be better?

```c
if (stream==stdin) return EBADF
if(stream==stdout || stream==stderr) ... clone
```

Reply:
If any of the used functions fails, we will get a `NULL` for the `FILE*`. What to do in that case? Keep the old setting?

Reply:
Can you think of anything more sensible than keeping the previous `FILE*`, and returning non-zero?

Reply:
> But `fclose()` may not like it ...

I have found some claims that `fclose(NULL)` is not portable.  Some (older?) glibc triggers [`*NULL`](https://stackoverflow.com/questions/16922871/why-glibcs-fclosenull-cause-segmentation-fault-instead-of-returning-error) and windows may popup [abort/retry/fail](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fclose-fcloseall?view=msvc-170).